src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -422,9 +422,16 @@ public BigInteger(ReadOnlySpan<byte> value, bool isUnsigned = false, bool isBigE
                     NumericsHelpers.DangerousMakeTwosComplement(val); // Mutates val
 
                     // Pack _bits to remove any wasted space after the twos complement
-                    int len = val.Length - 1;
-                    while (len >= 0 && val[len] == 0) len--;
-                    len++;
+                    int len = val.Length;
+                    if (val[len - 1] == 0)
+                    {
+                        len--;
+                        if (len > 0 && val[len - 1] == 0)
+                        {
+                            len = val.AsSpan().LastIndexOfAnyExcept((uint)0);
```

Comment:
Shouldn't we slice out the last 2 elements that already checked?

Reply:
I don't think so.
While that would be faster for some values, the cost of the slice would be for all values.

Since `LastIndexOfAnyExcept` effectively checks batches of 8. Slicing 2 elements should make 2 out of every 8 faster (land in an earlier batch), but at the cost of making the other 6 slower.
Each individual case seems to be either 0.2-0.3 ns slower, or 0.3-0.4 ns faster (with 6 slower cases for every 2 faster).

Compared current pr code with one line replaced:
`len = val.AsSpan().Slice(0, len - 1).LastIndexOfAnyExcept((uint)0);`

<details>
<summary>Details about benchmarks</summary>

BenchmarkDotNet=v0.13.5, OS=ubuntu 22.04 (container)
AMD EPYC 7763, 1 CPU, 4 logical and 2 physical cores
.NET SDK=8.0.100-preview.7.23376.3
  [Host]     : .NET 8.0.0 (8.0.23.37506), X64 RyuJIT AVX2
  DefaultJob : .NET 8.0.0 (8.0.23.37506), X64 RyuJIT AVX2


|           Method | NumberOfZeroElements |      Mean |     Error |    StdDev | Ratio | RatioSD | Allocated | Alloc Ratio |
|----------------- |--------------------- |----------:|----------:|----------:|------:|--------:|----------:|------------:|
|  SpanIndexOfOpt1 |                    0 | 0.3448 ns | 0.0386 ns | 0.0488 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                    0 | 0.3349 ns | 0.0304 ns | 0.0269 ns |  0.96 |    0.14 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                    1 | 0.6421 ns | 0.0457 ns | 0.0508 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                    1 | 0.8958 ns | 0.0501 ns | 0.0557 ns |  1.40 |    0.13 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                    2 | 2.8646 ns | 0.0905 ns | 0.0889 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                    2 | 3.1109 ns | 0.0739 ns | 0.0691 ns |  1.09 |    0.04 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                    3 | 2.8671 ns | 0.0903 ns | 0.0927 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                    3 | 3.1409 ns | 0.0968 ns | 0.1035 ns |  1.10 |    0.05 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                    4 | 2.8081 ns | 0.0876 ns | 0.0820 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                    4 | 3.1525 ns | 0.0968 ns | 0.0994 ns |  1.12 |    0.05 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                    5 | 2.7483 ns | 0.0539 ns | 0.0478 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                    5 | 3.0739 ns | 0.0902 ns | 0.0753 ns |  1.12 |    0.02 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                    6 | 2.8225 ns | 0.0830 ns | 0.0735 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                    6 | 3.0991 ns | 0.0610 ns | 0.0771 ns |  1.10 |    0.04 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                    7 | 2.7938 ns | 0.0778 ns | 0.0689 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                    7 | 3.0929 ns | 0.0832 ns | 0.0737 ns |  1.11 |    0.04 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                    8 | 3.4889 ns | 0.1021 ns | 0.1328 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                    8 | 3.1395 ns | 0.0961 ns | 0.1068 ns |  0.90 |    0.04 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                    9 | 3.5022 ns | 0.1018 ns | 0.1394 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                    9 | 3.1272 ns | 0.0954 ns | 0.1136 ns |  0.89 |    0.05 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   10 | 3.2514 ns | 0.1000 ns | 0.1027 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   10 | 3.5157 ns | 0.1030 ns | 0.0964 ns |  1.08 |    0.04 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   11 | 3.4852 ns | 0.0986 ns | 0.1012 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   11 | 3.5713 ns | 0.1031 ns | 0.1605 ns |  1.05 |    0.05 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   12 | 3.4929 ns | 0.1003 ns | 0.1194 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   12 | 3.5128 ns | 0.1027 ns | 0.1336 ns |  1.01 |    0.06 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   15 | 3.6601 ns | 0.1038 ns | 0.0867 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   15 | 4.0272 ns | 0.1075 ns | 0.1005 ns |  1.10 |    0.04 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   16 | 3.8484 ns | 0.1067 ns | 0.0998 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   16 | 3.4683 ns | 0.1003 ns | 0.1073 ns |  0.90 |    0.03 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   17 | 3.9508 ns | 0.0855 ns | 0.0714 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   17 | 3.5143 ns | 0.1004 ns | 0.1074 ns |  0.88 |    0.03 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   18 | 3.8507 ns | 0.1042 ns | 0.1070 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   18 | 4.1199 ns | 0.1135 ns | 0.1436 ns |  1.07 |    0.05 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   19 | 3.8129 ns | 0.0930 ns | 0.0777 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   19 | 4.2368 ns | 0.1097 ns | 0.0916 ns |  1.11 |    0.03 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   23 | 4.0141 ns | 0.1074 ns | 0.1149 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   23 | 4.1867 ns | 0.1161 ns | 0.1468 ns |  1.04 |    0.04 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   24 | 4.3750 ns | 0.1203 ns | 0.1126 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   24 | 4.0765 ns | 0.0614 ns | 0.0513 ns |  0.93 |    0.02 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   25 | 4.6308 ns | 0.1126 ns | 0.1464 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   25 | 4.1137 ns | 0.0632 ns | 0.0560 ns |  0.89 |    0.03 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   26 | 4.6234 ns | 0.1203 ns | 0.1181 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   26 | 4.4313 ns | 0.1214 ns | 0.1579 ns |  0.96 |    0.04 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   27 | 4.5395 ns | 0.1225 ns | 0.1458 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   27 | 4.6131 ns | 0.1249 ns | 0.1168 ns |  1.01 |    0.03 |         - |          NA |
|                  |                      |           |           |           |       |         |           |             |
|  SpanIndexOfOpt1 |                   28 | 4.5866 ns | 0.0866 ns | 0.0810 ns |  1.00 |    0.00 |         - |          NA |
| SpanIndexOfSlice |                   28 | 4.4707 ns | 0.1099 ns | 0.1028 ns |  0.98 |    0.03 |         - |          NA |
</details>